### ISSUE TITLE
gcc 8.3.0: Internal compiler error

### ISSUE THREAD

**Issue by neclepsio** [msys2/MINGW-packages#5006]
Compiling Qt5 fails with the following error. Downgrading to 8.2.0 solves the problem.

```
In file included from D:/Ignazio/IgnPack/Go/msys64/mingw64/include/QtCore/qcborstream.h:45,
                 from D:/Ignazio/IgnPack/Go/msys64/mingw64/include/QtCore/QtCore:40,
                 from D:/Ignazio/IgnPack/Go/msys64/mingw64/include/QtQml/QtQmlDepends:3,
                 from D:/Ignazio/IgnPack/Go/msys64/mingw64/include/QtQml/QtQml:3,
                 from cpp/capi.cpp:4,
                 from all.cpp:2:
D:/Ignazio/IgnPack/Go/msys64/mingw64/include/QtCore/qfloat16.h:79:54: internal compiler error: in make_rtl_for_nonlocal_decl, at cp/decl.c:6590
     Q_CORE_EXPORT static const quint32 mantissatable[];
                                                      ^
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://sourceforge.net/projects/msys2> for instructions.
```

**Comment by lazka** [comment]
Same bug in Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=923214

**Comment by Corruptinator** [comment]
Got the Same bug when I was trying to compile the open source video editor Olive:

```
In file included from C:/msys64/mingw64/include/QtCore/qcborstream.h:45,
                 from C:/msys64/mingw64/include/QtCore/QtCore:40,
                 from C:/msys64/mingw64/include/QtWidgets/QtWidgetsDepends:3,
                 from C:/msys64/mingw64/include/QtWidgets/QtWidgets:3,
                 from ui/flowlayout.cpp:52:
C:/msys64/mingw64/include/QtCore/qfloat16.h:79:54: internal compiler error: in make_rtl_for_nonlocal_decl, at cp/decl.c:6590
     Q_CORE_EXPORT static const quint32 mantissatable[];
                                                      ^
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://sourceforge.net/projects/msys2> for instructions.
make[1]: *** [Makefile.Debug:12506: debug/flowlayout.o] Error 1
make[1]: Leaving directory '/c/Github/olive'
make: *** [Makefile:38: debug] Error 2
```

**Comment by Corruptinator** [comment]
@neclepsio, do you happen to know how to downgrade to gcc 8.2?

**Comment by neclepsio** [comment]
Since I still had the files in the cache, I used:
```
cd /var/cache/pacman/pkg
pacman -U mingw-w64-x86_64-gcc-libs-8.2.0-3-any.pkg.tar.xz mingw-w64-x86_64-gcc-8.2.0-3-any.pkg.tar.xz
```

**Comment by Corruptinator** [comment]
Thanks!

**Comment by cbrt64** [comment]
I'm posting this in case it's helpful to anyone. The error-inducing code can be reduced to one line of preprocessed C++:

`class c { __attribute__((dllimport)) static int n; };`

**Comment by oscarfv** [comment]
There is a PR on https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88568 with two patches and a test case that resembles @cbrt64 example. The first patch was committed to gcc master and 8.x branches before the 8.3 release, but obviously it didn't solve the problem. The second patch was not yet committed because the author has no way of testing it (no Windows machine). Testing the simple case posted above is simple enough. More interesting would be to run a Qt build with the patched 8.3 gcc.

**Comment by Alexpux** [comment]
Apply patch from PR - https://github.com/msys2/MINGW-packages/commit/6d598c08d09ec9f94ceb6a167e04f47e4441a59b
